management/users/users.js

```diff
@@ -125,66 +136,181 @@ WDN.initializePlugin('modal', [function() {
+				});
+			};
+
+			function getUserInformation(userId, callbackFunction) {
+				$.ajax({
+					dataType: 'json',
+					method: 'GET',
+					url: '/server/management/users/users.php',
+					data: {
+						action: 'getUserInformation',
+						userId: userId
+					},
+					success: function(response) {
+						if (!response || !response.success) {
+							alert(response.error || 'There was an error fetching user data. Please refresh the page and try again.');
+							return;
+						}
+
+						callbackFunction({
+							'userId': userId,
+							'firstName': response.user.firstName,
+							'lastName': response.user.lastName,
+							'phoneNumber': response.user.phoneNumber,
+							'email': response.user.email
+						});
+					}
+				});
+			};
+
+			function updateUserInformation(userId, newFirstName, newLastName, newEmail, newPhoneNumber, callbackFunction) {
 				$.ajax({
 					dataType: 'json',
 					method: 'POST',
 					url: '/server/management/users/users.php',
 					data: {
-						callback: 'getUserTable'
+						action: 'updateUserInformation',
+						userId: userId,
+						firstName: newFirstName,
+						lastName: newLastName,
+						email: newEmail,
+						phoneNumber: newPhoneNumber
 					},
-					success: function(response){
-						if(response.success){
-							$('#user-management-table').html(response.table);
-							$('#user-management-table .userPermissionsSelectPicker').selectpicker({
-								iconBase: '',
-								tickIcon: 'wdn-icon-ok',
-								multipleSeparator: ', <br>'
-							});
-							$('#user-management-table .userAbilitiesSelectPicker').selectpicker({
-								iconBase: '',
-								tickIcon: 'wdn-icon-ok',
-								multipleSeparator: ', <br>'
-							});
-							$('.dropdown-toggle').on('click', function(){
-								// Set up event listener for clicks outside of the element
-								$(document).on('click.dropdown', function(e){
-									var container = $('.bootstrap-select.open');
-									// If the click is not on a child element
-									if (container.has(e.target).length === 0) {
-										container.removeClass('open');
-										$(document).off('click.dropdown');
-									}
-								});
-								// Close all dropdowns
-								$('.bootstrap-select.open').removeClass('open');
-								// Open this dropdown
-								$(this).parent().toggleClass('open');
-							});
-							$('.dropdown-toggle').siblings('select').hide();
-						}else{
-							alert(response.error);
+					success: function(response) {
+						if (!response || !response.success) {
+							alert(response.error || 'There was an error fetching the data. Please refresh the page.');
 						}
+
+						callbackFunction();
+						refreshUserTable();
+					}
+				});
+			};
+
+			function refreshUserTable() {
+				$.ajax({
+					dataType: 'json',
+					method: 'POST',
+					url: '/server/management/users/users.php',
+					data: {
+						action: 'getUserTable'
+					},
+					success: function(response) {
+						if (!response || !response.success) {
```

Comment:
You do a ton of these checks, might be worth making a helper function to put this logic in one place.
```
checkSuccessful(response, errorMessage = 'There was an error fetching the data. Please refresh the page.') {
    isSuccessful = response && response.success; // Or whatever it is
    if (!isSuccessful) {
        notify(errorMessage);
    }
    return isSuccessful;
}
```